ptx/lib/zluda_ptx_impl.cpp

```diff
@@ -1406,17 +1436,21 @@ extern "C"
                     if (specifier_with_length)
                     {
                         s += specifier_with_length;
-                    }
-                    if (len > 0)
-                    {
-                        uint64_t value = read_valist(valist_ptr, valist_offset, len);
-                        handle = __ockl_printf_append_args(handle, 1, value, 0, 0, 0, 0, 0, 0, 0);
+                    } else {
+                        s++;
+                        continue;
                     }
                 }
+
+                if (len > 0)
+                {
+                    uint64_t value = read_valist(valist_ptr, valist_offset, len);
+                    handle = __ockl_printf_append_args(handle, 1, value, 0, 0, 0, 0, 0, 0, 0);
+                }
+                break;
             }
         }
         return (uint32_t)__ockl_printf_append_args(handle, 0, 0, 0, 0, 0, 0, 0, 0, 1);
-        return 1;

```

Comment:
Please change so it does not return `__ockl_printf_append_args`, but 1 instead. `__ockl_printf_append_args` returns an arbitrary handle. I don't know what I were thinking